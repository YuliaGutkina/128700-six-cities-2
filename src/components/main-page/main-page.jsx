import React from 'react';
import PropTypes from "prop-types";
import {connect} from "react-redux";
import classNames from "classnames";

import OffersList from "../offers-list/offers-list";
import {CityMap} from "../city-map/city-map";
import Sorting from "../sorting/sorting";
import Header from "../header/header";
import Locations from "../locations/locations";
import withActiveItem from "../../hocs/with-active-item/with-active-item";
import {
  getActiveOfferSelector,
  receiveCityInfoSelector,
  receiveCityOffersSortedSelector
} from "../../reducer/data/selectors";


const OffersListWrapped = withActiveItem(OffersList);

const MainPage = (props) => {
  const {city, cityOffers, activeOffer} = props;
  const cityName = city ? city.name : ``;

  return <div className="page page--gray page--main">
    <Header/>
    <main className={classNames(`page__main page__main--index`, {
      "page__main--index-empty": !cityOffers.length
    })}>
      <h1 className="visually-hidden">Cities</h1>
      <Locations
        currentCity={cityName}
      />
      <div className="cities">
        {cityOffers.length ?
          <div className="cities__places-container container">
            <section className="cities__places places">
              <h2 className="visually-hidden">Places</h2>
              <b className="places__found">{cityOffers.length} places to stay in {cityName}</b>
              <Sorting/>
              <div className="cities__places-list places__list tabs__content">
                <OffersListWrapped
                  places={cityOffers}
                  cardClassName="cities__place-card"
                  imageWrapperClassName="cities__image-wrapper"
                  imageWidth={260}
                  imageHeight={200}
                />
              </div>
            </section>
            <div className="cities__right-section">
              <CityMap
                items={cityOffers}
                currentCity={city}
                activeItem={activeOffer}
              />
            </div>
          </div> :
          <div className="cities__places-container cities__places-container--empty container">
            <section className="cities__no-places">
              <div className="cities__status-wrapper tabs__content">
                <b className="cities__status">No places to stay available</b>
                <p className="cities__status-description">We could not find any property availbale at the moment in {cityName}</p>
              </div>
            </section>
            <div className="cities__right-section" />
          </div>
        }
      </div>
    </main>
  </div>;
};

MainPage.propTypes = {
  city: PropTypes.object,
  citiesList: PropTypes.array,
  cityOffers: PropTypes.array,
  activeOffer: PropTypes.object,
};

const mapStateToProps = (state, ownProps) => Object.assign({}, ownProps, {
  city: receiveCityInfoSelector(state),
  cityOffers: receiveCityOffersSortedSelector(state),
  activeOffer: getActiveOfferSelector(state)
});


export {MainPage};
export default connect(mapStateToProps)(MainPage);
