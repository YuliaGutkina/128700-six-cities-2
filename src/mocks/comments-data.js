export const commentsData = [
  {
    id: 1,
    user: {
      id: 4,
      isPro: false,
      name: `Max`,
      avatar: `img/1.png`
    },
    rating: 4,
    comment: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
    date: `2019-05-08T14:13:56.569Z`
  },
  {
    id: 2,
    user: {
      id: 4,
      isPro: false,
      name: `Max`,
      avatar: `img/1.png`
    },
    rating: 4,
    comment: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
    date: `2019-05-08T14:13:56.569Z`
  },
  {
    id: 3,
    user: {
      id: 4,
      isPro: false,
      name: `Max`,
      avatar: `img/1.png`
    },
    rating: 4,
    comment: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
    date: `2019-05-08T14:13:56.569Z`
  }
];
