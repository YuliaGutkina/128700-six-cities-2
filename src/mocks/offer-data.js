export const offersData = [
  {
    id: 1,
    city: {
      name: `Amsterdam`,
      location: {
        latitude: 52.370216,
        longitude: 4.895168,
        zoom: 10
      }
    },
    preview: `img/1.png`,
    images: [`img/1.png`, `img/2.png`],
    title: `Beautiful & luxurious studio at great location`,
    isFavorite: false,
    isPremium: false,
    rating: 4.8,
    type: `apartment`,
    bedrooms: 3,
    maxAdults: 4,
    price: 120,
    goods: [`Heating`, `Kitchen`, `Cable TV`, `Washing machine`, `Coffee machine`, `Dishwasher`],
    host: {
      id: 3,
      isPro: true,
      name: `Angelina`,
      avatar: `img/1.png`
    },
    description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
    location: {
      latitude: 52.35514938496378,
      longitude: 4.673877537499948,
      zoom: 8
    }
  },
  {
    id: 2,
    city: {
      name: `Amsterdam`,
      location: {
        latitude: 52.370216,
        longitude: 4.895168,
        zoom: 10
      }
    },
    preview: `img/1.png`,
    images: [`img/1.png`, `img/2.png`],
    title: `Beautiful & luxurious studio at great location`,
    isFavorite: false,
    isPremium: false,
    rating: 4.8,
    type: `apartment`,
    bedrooms: 3,
    maxAdults: 4,
    price: 120,
    goods: [`Heating`, `Kitchen`, `Cable TV`, `Washing machine`, `Coffee machine`, `Dishwasher`],
    host: {
      id: 3,
      isPro: true,
      name: `Angelina`,
      avatar: `img/1.png`
    },
    description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
    location: {
      latitude: 52.35514938496378,
      longitude: 4.673877537499948,
      zoom: 8
    }
  },
  {
    id: 3,
    city: {
      name: `Tel Aviv`,
      location: {
        latitude: 52.370216,
        longitude: 4.895168,
        zoom: 10
      }
    },
    preview: `img/1.png`,
    images: [`img/1.png`, `img/2.png`],
    title: `Beautiful & luxurious studio at great location`,
    isFavorite: false,
    isPremium: false,
    rating: 4.8,
    type: `apartment`,
    bedrooms: 3,
    maxAdults: 4,
    price: 120,
    goods: [`Heating`, `Kitchen`, `Cable TV`, `Washing machine`, `Coffee machine`, `Dishwasher`],
    host: {
      id: 3,
      isPro: true,
      name: `Angelina`,
      avatar: `img/1.png`
    },
    description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
    location: {
      latitude: 52.35514938496378,
      longitude: 4.673877537499948,
      zoom: 8
    }
  },
  {
    id: 4,
    city: {
      name: `Amsterdam`,
      location: {
        latitude: 52.370216,
        longitude: 4.895168,
        zoom: 10
      }
    },
    preview: `img/1.png`,
    images: [`img/1.png`, `img/2.png`],
    title: `Beautiful & luxurious studio at great location`,
    isFavorite: false,
    isPremium: false,
    rating: 4.8,
    type: `apartment`,
    bedrooms: 3,
    maxAdults: 4,
    price: 120,
    goods: [`Heating`, `Kitchen`, `Cable TV`, `Washing machine`, `Coffee machine`, `Dishwasher`],
    host: {
      id: 3,
      isPro: true,
      name: `Angelina`,
      avatar: `img/1.png`
    },
    description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
    location: {
      latitude: 52.35514938496378,
      longitude: 4.673877537499948,
      zoom: 8
    }
  },
  {
    id: 5,
    city: {
      name: `Amsterdam`,
      location: {
        latitude: 52.370216,
        longitude: 4.895168,
        zoom: 10
      }
    },
    preview: `img/1.png`,
    images: [`img/1.png`, `img/2.png`],
    title: `Beautiful & luxurious studio at great location`,
    isFavorite: false,
    isPremium: false,
    rating: 4.8,
    type: `apartment`,
    bedrooms: 3,
    maxAdults: 4,
    price: 120,
    goods: [`Heating`, `Kitchen`, `Cable TV`, `Washing machine`, `Coffee machine`, `Dishwasher`],
    host: {
      id: 3,
      isPro: true,
      name: `Angelina`,
      avatar: `img/1.png`
    },
    description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
    location: {
      latitude: 52.35514938496378,
      longitude: 4.673877537499948,
      zoom: 8
    }
  },
  {
    id: 6,
    city: {
      name: `Amsterdam`,
      location: {
        latitude: 52.370216,
        longitude: 4.895168,
        zoom: 10
      }
    },
    preview: `img/1.png`,
    images: [`img/1.png`, `img/2.png`],
    title: `Beautiful & luxurious studio at great location`,
    isFavorite: false,
    isPremium: false,
    rating: 4.8,
    type: `apartment`,
    bedrooms: 3,
    maxAdults: 4,
    price: 120,
    goods: [`Heating`, `Kitchen`, `Cable TV`, `Washing machine`, `Coffee machine`, `Dishwasher`],
    host: {
      id: 3,
      isPro: true,
      name: `Angelina`,
      avatar: `img/1.png`
    },
    description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
    location: {
      latitude: 52.35514938496378,
      longitude: 4.673877537499948,
      zoom: 8
    }
  },
  {
    id: 7,
    city: {
      name: `Amsterdam`,
      location: {
        latitude: 52.370216,
        longitude: 4.895168,
        zoom: 10
      }
    },
    preview: `img/1.png`,
    images: [`img/1.png`, `img/2.png`],
    title: `Beautiful & luxurious studio at great location`,
    isFavorite: false,
    isPremium: false,
    rating: 4.8,
    type: `apartment`,
    bedrooms: 3,
    maxAdults: 4,
    price: 120,
    goods: [`Heating`, `Kitchen`, `Cable TV`, `Washing machine`, `Coffee machine`, `Dishwasher`],
    host: {
      id: 3,
      isPro: true,
      name: `Angelina`,
      avatar: `img/1.png`
    },
    description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
    location: {
      latitude: 52.35514938496378,
      longitude: 4.673877537499948,
      zoom: 8
    }
  },
  {
    id: 8,
    city: {
      name: `Paris`,
      location: {
        latitude: 52.370216,
        longitude: 4.895168,
        zoom: 10
      }
    },
    preview: `img/1.png`,
    images: [`img/1.png`, `img/2.png`],
    title: `Beautiful & luxurious studio at great location`,
    isFavorite: false,
    isPremium: false,
    rating: 4.8,
    type: `apartment`,
    bedrooms: 3,
    maxAdults: 4,
    price: 120,
    goods: [`Heating`, `Kitchen`, `Cable TV`, `Washing machine`, `Coffee machine`, `Dishwasher`],
    host: {
      id: 3,
      isPro: true,
      name: `Angelina`,
      avatar: `img/1.png`
    },
    description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
    location: {
      latitude: 52.35514938496378,
      longitude: 4.673877537499948,
      zoom: 8
    }
  },
  {
    id: 9,
    city: {
      name: `Paris`,
      location: {
        latitude: 52.370216,
        longitude: 4.895168,
        zoom: 10
      }
    },
    preview: `img/1.png`,
    images: [`img/1.png`, `img/2.png`],
    title: `Beautiful & luxurious studio at great location`,
    isFavorite: false,
    isPremium: false,
    rating: 4.8,
    type: `apartment`,
    bedrooms: 3,
    maxAdults: 4,
    price: 120,
    goods: [`Heating`, `Kitchen`, `Cable TV`, `Washing machine`, `Coffee machine`, `Dishwasher`],
    host: {
      id: 3,
      isPro: true,
      name: `Angelina`,
      avatar: `img/1.png`
    },
    description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
    location: {
      latitude: 52.35514938496378,
      longitude: 4.673877537499948,
      zoom: 8
    }
  }
];
