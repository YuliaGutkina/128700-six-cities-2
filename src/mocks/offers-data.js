export const offersData = [
  {
    city: `Amsterdam`,
    initialCoordinates: [52.38333, 4.9],
    places: [
      {
        title: `Beautiful & luxurious apartment at great location`,
        type: `Apartment`,
        isPremium: true,
        src: `img/apartment-01.jpg`,
        price: 120,
        rating: 93,
        coordinates: [52.3909553943508, 4.85309666406198]
      },
      {
        title: `Wood and stone place`,
        type: `Private room`,
        isPremium: false,
        src: `img/room.jpg`,
        price: 80,
        rating: 80,
        coordinates: [52.369553943508, 4.85309666406198]
      },
      {
        title: `Canal View Prinsengracht`,
        type: `Apartment`,
        isPremium: false,
        src: `img/apartment-02.jpg`,
        price: 132,
        rating: 80,
        coordinates: [52.3909553943508, 4.929309666406198]
      },
      {
        title: `Nice, cozy, warm big bed apartment`,
        type: `Apartment`,
        isPremium: true,
        src: `img/apartment-03.jpg`,
        price: 180,
        rating: 100,
        coordinates: [52.3809553943508, 4.939309666406198]
      }
    ]
  },
  {
    city: `Paris`,
    initialCoordinates: [48.864716, 2.349014],
    places: []
  },
  {
    city: `Hamburg`,
    initialCoordinates: [53.551086, 9.993682],
    places: []
  }
];
