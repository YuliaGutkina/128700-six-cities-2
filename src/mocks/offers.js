export const offers = [
  {
    title: `Beautiful & luxurious apartment at great location`,
    type: `Apartment`,
    isPremium: true,
    src: `img/apartment-01.jpg`,
    price: 120,
    rating: 93
  },
  {
    title: `Wood and stone place`,
    type: `Private room`,
    isPremium: false,
    src: `img/room.jpg`,
    price: 80,
    rating: 80
  },
  {
    title: `Canal View Prinsengracht`,
    type: `Apartment`,
    isPremium: false,
    src: `img/apartment-02.jpg`,
    price: 132,
    rating: 80
  },
  {
    title: `Nice, cozy, warm big bed apartment`,
    type: `Apartment`,
    isPremium: true,
    src: `img/apartment-03.jpg`,
    price: 180,
    rating: 100
  }
];
