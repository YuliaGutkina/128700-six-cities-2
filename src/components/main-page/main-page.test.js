import React from "react";
import ShallowRenderer from 'react-test-renderer/shallow';

import {MainPage} from "./main-page";
import {offersData} from "../../mocks/offers-data";

it(`Main page correctly renders after relaunch`, () => {
  const renderer = new ShallowRenderer();
  renderer.render(
      <MainPage
        offersData={offersData}
        city="Amsterdam"
        cityOffers={{
          city: `Amsterdam`,
          coordinates: [0, 0],
          places: [{}, {}]
        }}
        onCardTitleClick={jest.fn()}
      />
  );
  const result = renderer.getRenderOutput();

  expect(result).toMatchSnapshot();
});
