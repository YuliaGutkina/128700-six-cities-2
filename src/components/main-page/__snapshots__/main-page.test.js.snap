// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main page correctly renders after relaunch 1`] = `
<main
  className="page__main page__main--index"
>
  <h1
    className="visually-hidden"
  >
    Cities
  </h1>
  <Memo(Connect(Locations))
    currentCity="Amsterdam"
    offers={
      Array [
        Object {
          "city": "Amsterdam",
          "initialCoordinates": Array [
            52.38333,
            4.9,
          ],
          "places": Array [
            Object {
              "coordinates": Array [
                52.3909553943508,
                4.85309666406198,
              ],
              "isPremium": true,
              "price": 120,
              "rating": 93,
              "src": "img/apartment-01.jpg",
              "title": "Beautiful & luxurious apartment at great location",
              "type": "Apartment",
            },
            Object {
              "coordinates": Array [
                52.369553943508,
                4.85309666406198,
              ],
              "isPremium": false,
              "price": 80,
              "rating": 80,
              "src": "img/room.jpg",
              "title": "Wood and stone place",
              "type": "Private room",
            },
            Object {
              "coordinates": Array [
                52.3909553943508,
                4.929309666406198,
              ],
              "isPremium": false,
              "price": 132,
              "rating": 80,
              "src": "img/apartment-02.jpg",
              "title": "Canal View Prinsengracht",
              "type": "Apartment",
            },
            Object {
              "coordinates": Array [
                52.3809553943508,
                4.939309666406198,
              ],
              "isPremium": true,
              "price": 180,
              "rating": 100,
              "src": "img/apartment-03.jpg",
              "title": "Nice, cozy, warm big bed apartment",
              "type": "Apartment",
            },
          ],
        },
        Object {
          "city": "Paris",
          "initialCoordinates": Array [
            48.864716,
            2.349014,
          ],
          "places": Array [],
        },
        Object {
          "city": "Hamburg",
          "initialCoordinates": Array [
            53.551086,
            9.993682,
          ],
          "places": Array [],
        },
      ]
    }
  />
  <div
    className="cities"
  >
    <div
      className="cities__places-container container"
    >
      <section
        className="cities__places places"
      >
        <h2
          className="visually-hidden"
        >
          Places
        </h2>
        <b
          className="places__found"
        >
          2
           places to stay in 
          Amsterdam
        </b>
        <form
          action="#"
          className="places__sorting"
          method="get"
        >
          <span
            className="places__sorting-caption"
          >
            Sort by
          </span>
          <span
            className="places__sorting-type"
            tabIndex={0}
          >
            Popular
            <svg
              className="places__sorting-arrow"
              height={4}
              width={7}
            >
              <use
                xlinkHref="#icon-arrow-select"
              />
            </svg>
          </span>
          <ul
            className="places__options places__options--custom places__options--opened"
          >
            <li
              className="places__option places__option--active"
              tabIndex={0}
            >
              Popular
            </li>
            <li
              className="places__option"
              tabIndex={0}
            >
              Price: low to high
            </li>
            <li
              className="places__option"
              tabIndex={0}
            >
              Price: high to low
            </li>
            <li
              className="places__option"
              tabIndex={0}
            >
              Top rated first
            </li>
          </ul>
        </form>
        <OffersList
          onCardTitleClick={[MockFunction]}
          places={
            Array [
              Object {},
              Object {},
            ]
          }
        />
      </section>
      <div
        className="cities__right-section"
      >
        <CityMap
          initialCity={
            Object {
              "coordinates": undefined,
              "name": "Amsterdam",
            }
          }
          items={
            Array [
              Object {},
              Object {},
            ]
          }
        />
      </div>
    </div>
  </div>
</main>
`;
